### ISSUE TITLE
ambiguous column name in ManyToMany query

### ISSUE THREAD

**Issue by randomiser** [IlyaSemenov/objection-graphql-resolver#7]
hey Ilya,

Been experimenting and everything looks to be working well although I’m having problems retrieving graphql data via Model. ManyToManyRelatation relationships. 

The actual query generated seems to include extra fields that aren’t prefixed correctly and as such won’t execute. 

Do you know if these kinds of relationships work currently? When I try making the equivalent queries using objection outside of graphql they return as I’d expect. 

Don’t have access to my code/SQL right now but can attach it later if it would be useful. 

Anyways, Hope you can answer that for me. 

Thanks again for building and releasing this project. Making my transition from rest pretty painless. 

Kind regards,
Matt

**Comment by IlyaSemenov** [comment]
To be honest, I never used many-to-many relations in objection, so there might be a shortcoming indeed. I'll have a look at this.

**Comment by randomiser** [comment]
Ok cool thanks. Appreciate it.

**Comment by IlyaSemenov** [comment]
I added a simple [test for m2m query](https://github.com/IlyaSemenov/objection-graphql-resolver/blob/master/tests/m2m.ts) and it seems to be working fine. Please come up with a failing example.

**Comment by randomiser** [comment]
Thanks! I’ll see what I’m doing that might be different. I’m away for the next couple of days so will get back to you once I’m back. Appreciate your help and that sample test case.

**Comment by randomiser** [comment]
Hi Ilya,

Found out what the issue seems to be.  By default all of my tables all have a primary key of 'id' including many to many tables.

If I add a primary key called id to the "book_author_rel" example above and rerun the tests I get this issue:

 FAIL ​ tests/m2m.ts 1 failed of 1 352.89ms
 ✖select `author_book_rel`.`book_id` as `objectiontmpjoin0`, `id`, `name` from `author` inner join `author_book_rel` on
  `author`.`id` = `author_book_rel`.`author_id` where `author_book_rel`.`book_id` in (1, 2, 3) - SQLITE_ERROR: ambiguous
  column name: id: {"response":{"errors":[{"message":"select `author_book_rel`.`book_id` as `objectiontmpjoin0`, `id`,
  `name` from `author` inner join `author_book_rel` on `author`.`id` = `author_book_rel`.`author_id` where
  `author_book_rel`.`book_id` in (1, 2, 3) - SQLITE_ERROR: ambiguous column name:
  id","locations":[{"line":2,"column":3}],"path":["library"],"extensions":{"code":"INTERNAL_SERVER_ERROR","exception":{"n
  ame":"DBError","nativeError":{"errno":1,"code":"SQLITE_ERROR"},"client":"sqlite","stacktrace":["DBError: select
  `author_book_rel`.`book_id` as `objectiontmpjoin0`, `id`, `name` from `author` inner join `author_book_rel` on
  `author`.`id` = `author_book_rel`.`author_id` where `author_book_rel`.`book_id` in (1, 2, 3) - SQLITE_ERROR: ambiguous
  column name: id","    at wrapError
  (/Examples/objection-graphql-resolver-master/node_modules/db-errors/lib/dbErrors.js:19:14)","    at
  handleExecuteError (/Examples/objection-graphql-resolver-master/node_modules/objection/lib/queryBuilder/Que
  ryBuilder.js:1123:32)","    at QueryBuilder.execute (/Examples/objection-graphql-resolver-master/node_modul
  es/objection/lib/queryBuilder/QueryBuilder.js:449:20)","    at WhereInEagerOperation.fetchRelation
  (/Examples/objection-graphql-resolver-master/node_modules/objection/lib/queryBuilder/operations/eager/Where
  InEagerOperation.js:100:20)","    at WhereInEagerOperation.onAfter2 (/Examples/objection-graphql-resolver-m
  aster/node_modules/objection/lib/queryBuilder/operations/eager/WhereInEagerOperation.js:70:5)","    at
  QueryBuilder.callAsyncOperationMethod (/Examples/objection-graphql-resolver-master/node_modules/objection/l
  ib/queryBuilder/QueryBuilderOperationSupport.js:387:14)","    at QueryBuilder.execute (/Examples/objection-
  graphql-resolver-master/node_modules/objection/lib/queryBuilder/QueryBuilder.js:447:14)","    at async Promise.all
  (index 0)"]}}}],"data":null,"status":200},"request":{"query":"{\n  library {\n    id\n    title\n    authors {\n
  id\n      name\n    }\n  }\n}\n"}}

**Comment by IlyaSemenov** [comment]
Hmm... That is what you will get with raw objection query as well:

```ts
await BookModel.query()
  .withGraphFetched("authors(authors)")
  .modifiers({
    authors: (authors) => authors.select("id"),
  })
```

```
✖ select `author_book_rel`.`book_id` as `objectiontmpjoin0`, `id` from `author` inner join `author_book_rel` on `author`.`id` = `author_book_rel`.`author_id` where
   `author_book_rel`.`book_id` in (1, 2, 3) - SQLITE_ERROR: ambiguous column name: id
```

But if one can work around that in raw queries with manual prefixing, the graph resolver must be able to infer the need for it automatically indeed. Thank you for clearing the problem.

**Comment by IlyaSemenov** [comment]
This was fixed in version 4.1.2.
